helix-rest/src/main/java/org/apache/helix/rest/server/resources/lock/LockAccessor.java

```diff
@@ -0,0 +1,228 @@
+            lockScope = generateHelixLockScope(mapFields.get(lockScopeKey.name()));
+          }
+        }
+      }
+      if (lockScope == null) {
+        throw new Exception(
+            "Failed to parse a lock scope. Please check the lock scope property and path keys and make sure they are correct.");
+      }
+      return new LockLocator(zkAddr, lockScope.getPath());
+    }
+
+    return new LockLocator(zkAddr, lockPath);
+  }
+
+  private LockInfo parseLockInfo(ZNRecord znRecord) throws Exception {
+    if (znRecord == null) {
+      return null;
+    }
+    // Parse user iod
+    String userId = znRecord.getSimpleField(USER_ID);
+
+    // Parse lock timeout
+    long lockTimeout;
+    try {
+      lockTimeout = Long.parseLong(znRecord.getSimpleField(LOCK_TIMEOUT));
+    } catch (NumberFormatException e) {
+      throw new Exception("Please check the timeout value input, could not convert to long.");
+    }
+
+    // Parse lock message
+    String lockMsg = znRecord.getSimpleField(LOCK_MSG);
+    return new LockInfo(userId, lockMsg, lockTimeout);
+  }
+
+  private LockScope generateHelixLockScope(Map<String, String> lockScopeVal) throws Exception {
+    // Get the lock scope property: cluster, participant, resource, etc.
+    String CLUSTER = "cluster";
+    String RESOURCE = "resource";
+    String PARTICIPANT = "participant";
+    String lockScopeProperty = lockScopeVal.get(LOCK_SCOPE_PROPERTY);
+    HelixLockScope.LockScopeProperty property;
+    try {
+      property = HelixLockScope.LockScopeProperty.valueOf(lockScopeProperty);
+    } catch (IllegalArgumentException e) {
+      throw new Exception(String.format(
+          "User requested lock scope property %s, but this property does not exist in HelixLockScope.",
+          lockScopeProperty));
+    }
+
+    // Parse the path keys to construct the lock path
+    List<String> pathKeys = Lists.newArrayList();
+    switch (property) {
+      case CLUSTER: {
+        String clusterName = lockScopeVal.get(CLUSTER);
+        if (clusterName != null) {
+          pathKeys.add(clusterName);
+        }
+      }
+      break;
+      case RESOURCE: {
+        String clusterName = lockScopeVal.get(CLUSTER);
+        String resourceName = lockScopeVal.get(RESOURCE);
+        if (clusterName != null && resourceName != null) {
+          pathKeys.add(clusterName);
+          pathKeys.add(resourceName);
+        }
+      }
+      break;
+      case PARTICIPANT: {
+        String clusterName = lockScopeVal.get(CLUSTER);
+        String participantName = lockScopeVal.get(PARTICIPANT);
+        if (clusterName != null && participantName != null) {
+          pathKeys.add(clusterName);
+          pathKeys.add(participantName);
+        }
+      }
+      default:
+        // Not gonna happen
+    }
+    if (pathKeys.isEmpty()) {
+      throw new Exception(
+          "Not enough path keys are provided to construct the lock path. Please check the request.");
```

Comment:
Let the user know what the request was. Also do we need to log it on the server side?